src/components/NewsGrid/index.jsx

```diff
@@ -2,17 +2,59 @@ import React from 'react';
 import {SimpleGrid} from '@chakra-ui/react';
 import NewsCard from "../NewsCard";
 import {useStockNews} from "../../contexts/useStockNews";
+import {DragDropContext, Droppable, Draggable} from "react-beautiful-dnd"
 
 const NewsGrid = () => {
-
     const {allNewsSources} = useStockNews();
+    const {resetNews} = useStockNews();
+
+    const reorder = (list, startIndex, endIndex) => {
+        const result = [...list];
+        const [removed] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removed);
+        return result;
+    }
 
     return (
-        <SimpleGrid columns={{base: 1, sm: 2, md: 3, lg: 4}} spacing={3}>
-            {
-                allNewsSources.map(newsSource => <NewsCard key={newsSource} source={newsSource}/> )
+        <DragDropContext onDragEnd={(result) => {
+            const {source, destination} = result;
+            if (!destination) {
+                return;
+            }
+            if (source.index === destination.index && source.droppableId === destination.droppableId) {
+                return;
             }
-        </SimpleGrid>
+            resetNews(reorder(allNewsSources, source.index, destination.index))
+        }}>
+            <Droppable droppableId="newsGrid">
+                {(droppableProvided) => (
+                    <div
```

Comment:
Not sure if we need div's here and if chakra ui's inbuilt components won't work

Reply:
Ah I had to add the divs eventually since _you may not use the ref attribute on function components because they don’t have instances_ (https://reactjs.org/docs/refs-and-the-dom.html#refs-and-function-components)

I tried implementing some of the alternative approaches presented in that doc, and here: https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/guides/using-inner-ref.md#exposing-a-dom-ref-from-your-component-

but I couldn't get them to work as needed, so I enveloped chakra's components in divs